src/bin/languageserver/mod.rs

```diff
@@ -1158,26 +1168,52 @@ impl LanguageServer for SolangServer {
     async fn did_open(&self, params: DidOpenTextDocumentParams) {
         let uri = params.text_document.uri;
 
+        if let Ok(path) = uri.to_file_path() {
```

Comment:
Is there nothing to do in the error case?

Reply:
the `did_open` method from the trait `tower_lsp::LanguageServer` trait doesn't return anything back to the client. 
So, if there is something wrong with the `uri` that the editor (VSCode in our case) sent us, we can't report it back. 

Moreover, because the `uri` is not built by us, but by the `tower_lsp` library itself, we should hardly encounter an error  here. Looking at the implementation of the function `to_file_path`, errors are returned when the `uri` isn't a valid URI, which is unlikely to occur given that they are always constructed by the same library, from where the `to_file_path` function originates.

Reply:
From what I understand, the `uri` is parsed by the `URL` crate. Just because it is a valid `URL` or `URI` does not mean it's a valid file path. It could be any kind of `URI`. For example, if we start using the the language server in web based editor like remix, it seems unlikely that file paths are used.

Should for some reason `.to_file_path()` fail, the current code silently swallows the error. The options we have are:
 - Silently ignore the error
 - panic
 - log a message

I think it is best if we log a message in this case.